interface/modules/custom_modules/oe-module-claimrev-connect/src/EraPage.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/**
+ *
+ * @package OpenEMR
+ * @link    http://www.open-emr.org
+ *
+ * @author    Brad Sharp <brad.sharp@claimrev.com>
+ * @copyright Copyright (c) 2022 Brad Sharp <brad.sharp@claimrev.com>
+ * @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ */
+
+namespace OpenEMR\Modules\ClaimRevConnector;
+
+use OpenEMR\Modules\ClaimRevConnector\EraSearch;
+
+class EraPage
+{
+    public static function searchEras($postData)
+    {
+        $startDate = $_POST['startDate'];
+        $endDate = $_POST['endDate'];
+        $fileStatus = $_POST['downloadStatus'];
+
+        
+        $model = new FileSearchModel();
+        $model->fileStatus = intval($fileStatus);
+        $model->ediType = "835";
+        $model->receivedDateStart = $startDate;
+        $model->receivedDateEnd = $endDate;
+
+        if ($model->receivedDateStart == "") {
+            $model->receivedDateStart = null;
+        }
+        if ($model->receivedDateEnd == "") {
+            $model->receivedDateEnd = null;
+        }
+        $data = EraSearch::search($model);
+        return $data;
+    }
+    public static function downloadEra($id)
+    {
+        $data = EraSearch::downloadEra($id);
+        $data->fileName = $data->ediType . "-" . $data->payerNumber . "-" . $id . ".txt";
```

Comment:
surround $id in above line with the convert_safe_file_dir_name() function to prevent directory walking (since this $id is being brought in by user via get parameter)

Reply:
Need to use convert_safe_file_dir_name() on the $id in line 43 and not in line 42